### ISSUE TITLE
BitmapText text width calculation incorrect

### ISSUE THREAD

**Issue by neymanushka** [pixijs/pixijs#7758]
BitmapText textWidth have different value for lines with same amount of symbols (used monospace font, xadvance is set to same value), broke after this pr #7479.

### Expected Behavior
textWidth for characters '1111' must be same as for '3333' (pixijs 6.0.4 or lower)
### Current Behavior
textWidth not same (pixijs 6.1.2)
### Possible Solution
i think lastLineWidth calculation is wrong

### Environment

- **`pixi.js` version**:  > 6.0.4
- **Browser & Version**:  Chrome/FF
- **OS & Version**: Windows
- **How is it supposed to work (pixi 6.0.4)**: https://jsfiddle.net/etrc786d/
- **How does it work now (pixi 6.1.3)**: https://jsfiddle.net/u38gvtqy/

**Comment by ShukantPal** [comment]
I wouldn't expect the width of "1111" to be the same as "3333". "3" is wider than "1".

**Comment by shumkovdenis** [comment]
> I wouldn't expect the width of "1111" to be the same as "3333". "3" is wider than "1".

If you use the monospace font, then the width of all characters is the same. For example, make a bitmap font based on https://fonts.google.com/specimen/Roboto+Mono?category=Monospace

**Comment by ShukantPal** [comment]
You’re right in the case of monospace fonts. Were you specifically talking about monospace fonts? If so, sorry for overlooking that part.

Sent from my iPad

On Aug 31, 2021, at 8:05 AM, Denis ***@***.***> wrote:

﻿

I wouldn't expect the width of "1111" to be the same as "3333". "3" is wider than "1".

If you use the monospace font, then the width of all characters is the same. For example, make a bitmap font based on https://fonts.google.com/specimen/Roboto+Mono?category=Monospace

**Comment by neymanushka** [comment]
> 
> 
> I wouldn't expect the width of "1111" to be the same as "3333". "3" is wider than "1".

yep,but only if not set xadvance. character is aligned to the length specified in this property

```
char id=48   x=45    y=0     width=13    height=20    xoffset=1     yoffset=6     xadvance=15    page=0  chnl=15
char id=49   x=129   y=0     width=9     height=20    xoffset=2     yoffset=6     xadvance=15    page=0  chnl=15

```
in this case all characters must be of the same length,i think.

**Comment by neymanushka** [comment]
first post updated, added more examples.

**Comment by kjarrio** [comment]
Here is a PR with a fix: #7933
